src/puppetlabs/puppetdb/query/reports.clj

```diff
@@ -105,6 +86,40 @@
    :logs
    :certname])
 
+(defn logs-query->sql
+  "Converts a vector-structured `query` to a corresponding SQL query which will
+  return nodes matching the `query`."
+  ([version query]
+   (logs-query->sql version query {}))
+  ([version query paging-options]
+   {:pre  [((some-fn nil? sequential?) query)]
+    :post [(map? %)
+           (jdbc/valid-jdbc-query? (:results-query %))
+           (or (not (:count? paging-options))
+               (jdbc/valid-jdbc-query? (:count-query %)))]}
+   (paging/validate-order-by! [:logs] paging-options)
+   (qe/compile-user-query->sql qe/report-logs-query query paging-options)))
+
+(defn munge-data-rows
```

Comment:
There are other munge functions in this namespace, we should move it to be near them, and make the naming less ambiguous (ie. munge-data-rows v munge-result-rows?).

Also ... no docstring.